google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/application/AppEngineAdminService.java

```diff
@@ -49,8 +73,54 @@ public static AppEngineAdminService getInstance() {
    * @throws IOException if there was a transient error connecting to the API
    */
   @Nullable
-  public abstract Application getApplicationForProjectId(@NotNull String projectId,
-      @NotNull Credential credential) throws IOException, GoogleApiException;
+  public Application getApplicationForProjectId(@NotNull String projectId,
```

Comment:
Fair points. The nice thing about this abstraction is that it lets us separate the caching logic from the actual business of communicating with the App Engine admin API. 

But perhaps this was a bad move since it breaks with the pattern we've been using throughout the codebase for these services. I'll move this code back into the implementation class.